jlm/hls/opt/InvariantLambdaMemoryStateRemoval.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2025 Magnus Själander <work@sjalander.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#include <jlm/hls/opt/InvariantLambdaMemoryStateRemoval.hpp>
+#include <jlm/llvm/ir/operators/MemoryStateOperations.hpp>
+#include <jlm/llvm/ir/operators/Phi.hpp>
+#include <jlm/rvsdg/lambda.hpp>
+#include <jlm/rvsdg/RvsdgModule.hpp>
+#include <jlm/rvsdg/traverser.hpp>
+#include <jlm/util/Statistics.hpp>
+
+namespace jlm::hls
+{
+
+void
+InvariantLambdaMemoryStateRemoval::RemoveInvariantMemoryStateEdges(
+    rvsdg::RegionResult * memoryState)
```

Comment:
Maybe you can call it `memoryStateResult` so one does know what exactly it is when reading the code.

Reply:
Changed